From 41f5294f1bfdfcbe96ed0d19ec79dc2f3cc24d36 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Thu, 20 May 2021 11:03:20 +0200
Subject: [PATCH] analogix plugin requires gusb
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Otherwise the following build error happens:

  ../fwupd-1.6.0/plugins/analogix/fu-analogix-device.c:54:10: error: ‘G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE’ undeclared (first use in this function)
     54 |          G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ../fwupd-1.6.0/plugins/analogix/fu-analogix-device.c:54:10: note: each undeclared identifier is reported only once for each function it appears in
  ../fwupd-1.6.0/plugins/analogix/fu-analogix-device.c:55:10: error: ‘G_USB_DEVICE_REQUEST_TYPE_VENDOR’ undeclared (first use in this function)
     55 |          G_USB_DEVICE_REQUEST_TYPE_VENDOR,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ../fwupd-1.6.0/plugins/analogix/fu-analogix-device.c:56:10: error: ‘G_USB_DEVICE_RECIPIENT_DEVICE’ undeclared (first use in this function)
     56 |          G_USB_DEVICE_RECIPIENT_DEVICE,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ../fwupd-1.6.0/plugins/analogix/fu-analogix-device.c: In function ‘fu_analogix_device_receive’:
  ../fwupd-1.6.0/plugins/analogix/fu-analogix-device.c:96:10: error: ‘G_USB_DEVICE_DIRECTION_DEVICE_TO_HOST’ undeclared (first use in this function)
     96 |          G_USB_DEVICE_DIRECTION_DEVICE_TO_HOST,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ../fwupd-1.6.0/plugins/analogix/fu-analogix-device.c:97:10: error: ‘G_USB_DEVICE_REQUEST_TYPE_VENDOR’ undeclared (first use in this function)
     97 |          G_USB_DEVICE_REQUEST_TYPE_VENDOR,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ../fwupd-1.6.0/plugins/analogix/fu-analogix-device.c:98:10: error: ‘G_USB_DEVICE_RECIPIENT_DEVICE’ undeclared (first use in this function)
     98 |          G_USB_DEVICE_RECIPIENT_DEVICE,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  and more...
---
 plugins/analogix/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plugins/analogix/meson.build b/plugins/analogix/meson.build
index 1bfbbc33..c112d72d 100644
--- a/plugins/analogix/meson.build
+++ b/plugins/analogix/meson.build
@@ -1,3 +1,4 @@
+if get_option('gusb')
 cargs = ['-DG_LOG_DOMAIN="FuPluginAnalogix"']
 
 install_data(['analogix.quirk'],
@@ -28,3 +29,4 @@ shared_module('fu_plugin_analogix',
     plugin_deps,
   ],
 )
+endif
-- 
2.32.0.rc0

